src/map/ai/controllers/trust_controller.cpp

```diff
@@ -209,6 +216,23 @@ bool CTrustController::Ability(uint16 targid, uint16 abilityid)
     return false;
 }
 
+bool CTrustController::Cast(uint16 targid, SpellID spellid)
+{
+    FaceTarget(targid);
+    if (static_cast<CMobEntity*>(POwner)->PRecastContainer->Has(RECAST_MAGIC, static_cast<uint16>(spellid)))
+    {
+        return false;
+    }
```

Comment:
I know there's a check in getHighest, but are there checks for selectSpecific for if the trust has enough MP?